lib/asset_checker.rb

```diff
@@ -0,0 +1,39 @@
+class AssetChecker
+  def self.run(argv)
+    assets_file = 'app/assets/javascripts/assets.js.erb'
+    translations_file = 'app/assets/javascripts/i18n-strings.js.erb'
+    @asset_strings = load_included_strings(assets_file)
+    @translation_strings = load_included_strings(translations_file)
+    argv.map { |f| check_file(f) }
+  end
+
+  def self.check_file(file)
+    data = File.open(file).read
+    missing_translations = find_missing(data, /\Wt\(['"](.*)['"]\)/, @translation_strings)
+    missing_assets = find_missing(data, /\WassetPath=["'](.*)['"]/, @asset_strings)
+    if missing_translations.any? || missing_assets.any? # rubocop:disable Style/GuardClause
+      warn file
+      missing_translations.each do |t|
+        warn "Missing translation, #{t}"
+      end
+      missing_assets.each do |a|
+        warn "Missing asset, #{a}"
+      end
+    end
+  end
+
+  def self.find_missing(file_data, pattern, source)
+    missing = []
+    strings = file_data.scan pattern
+    strings.each do |s|
+      missing.push(s) unless source.include? s
+    end
+    missing
```

Comment:
Might be a good use-case for `select`:

```suggestion
    strings = file_data.scan pattern
    strings.select { |s| ! source.include? s }
```

https://ruby-doc.org/core-2.4.1/Array.html#method-i-select

Reply:
I think a `reject` would be better, but I'm happy to online it if that's desirable. 

`strings.reject { |s| source.include? s }`

Reply:
`reject` is even better, I agree 👍